### PR TITLE
RHDEVDOCS-3549 - Update Forward to Logstash

### PR DESCRIPTION
**Pull request by libander**
- Aligned team: Dev Tools
- For branches: 4.8+
- Jira:https://issues.redhat.com/browse/RHDEVDOCS-3549
- Direct link to doc preview: https://deploy-preview-41946--osdocs.netlify.app/openshift-enterprise/latest/logging/cluster-logging-external.html#cluster-logging-collector-log-forward-fluentd_cluster-logging-external
- SME review: @alanconway 
- QE review: @anpingli
- Peer review: @rolfedh

### CODE REVIEW

**Comment by anpingli** [comment]
lgtm

**Comment by libander** [comment]
@anpingli - the preview displays some of the brackets oddly, are they correct?

File: modules/cluster-logging-collector-log-forward-fluentd.adoc [hunk 1]
```diff
@@ -81,3 +81,28 @@ spec:
 ----
 $ oc create -f <file-name>.yaml
 ----
+
+[id="cluster-logging-collector-log-forward-fluentd-config]
+== Enable nanosecond_precision for Logstash to ingest data from fluentd
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-fluentd.adoc:86]:
```suggestion
== Enabling nanosecond_precision for Logstash to ingest data from fluentd
```

File: modules/cluster-logging-collector-log-forward-fluentd.adoc [hunk 2]
```diff
@@ -81,3 +81,28 @@ spec:
...
+For Logstash to ingest log data from fluentd, you must set `{ nanosecond_precision => true }` for input in the Logstash configuration file shown here:
```
Comment (rolfedh) [inline modules/cluster-logging-collector-log-forward-fluentd.adoc:87]:
```suggestion
For Logstash to ingest log data from fluentd, you must set `{ nanosecond_precision => true }` for input in the Logstash configuration file shown here:

For Logstash to ingest log data from fluentd, you must enable nanosecond precision in the Logstash configuration file.

.Procedure

* In the Logstash configuration file,  set `{ nanosecond_precision => true }`.

.Example Logstash configuration file
```

**Review by rolfedh — APPROVED** [review]

**Review by anpingli — COMMENTED** [review]
/lgtm

File: modules/cluster-logging-collector-log-forward-fluentd.adoc [hunk 3]
```diff
@@ -81,3 +81,32 @@ spec:
...
+== Enabling nanosecond_precision for Logstash to ingest data from fluentd
+For Logstash to ingest log data from fluentd, you must enable nanosecond precision in the Logstash configuration file.
+
+.Procedure
+* In the Logstash configuration file,  set `{ nanosecond_precision => true }`.
+
+.Example Logstash configuration file
+[source,yaml]
+----
+  input {
```
Comment (alanconway) [inline modules/cluster-logging-collector-log-forward-fluentd.adoc:95]: I'm not familiar with logstash config, but this looks odd  - the indenting and line breaks are a bit random, it's not well formed YAML, it looks a bit like Ruby Hash syntax but it's not well formed for that either. Is this example correct?

Comment (libander) [reply modules/cluster-logging-collector-log-forward-fluentd.adoc:95]: @anpingli can you provide a sample logstash config file please? this one seems wrong.

Comment (anpingli) [reply modules/cluster-logging-collector-log-forward-fluentd.adoc:95]: @libander  The example is correct.  The logstash configuration file is not yam.

Comment (anpingli) [reply modules/cluster-logging-collector-log-forward-fluentd.adoc:95]: To be more clear, we can use content as below:

```
input { tcp { codec => fluent { nanosecond_precision => true } port => 24114 } }
filter { }
output { stdout { codec => rubydebug } }
```

**Review by anpingli — COMMENTED** [review]
/lgtm

File: modules/cluster-logging-collector-log-forward-fluentd.adoc (hunk 3 above)
Comment (alanconway) [reply modules/cluster-logging-collector-log-forward-fluentd.adoc]: +1 - I was confused by the `[source,yaml]` tag on line 93 - that should be changed to indicate the actual format,  or to just `[source]` if the doc generator doesn't know about the logstash format - it is not like any common config format that I recognize.

**Comment by libander** [comment]
https://github.com/openshift/openshift-docs/pull/47711

**Pull request closed without merging**